Guideline for LLM usage in bug reports (#14899)
ocaml/ocaml: AI.md

   of code that contributors do not understand; plagiarised works; hallucinated
   security reports; or experiments involving humans without their advance
   consent.

> ## Guidelines relevant to AI-assisted bug report

Reviewer: ```suggestion
## Guidelines relevant to AI-assisted bug reports
```

> Our advice is to avoid adding LLM-generated contents or explanation to

Reviewer 2: ```suggestion
Our advice is to avoid adding LLM-generated contents or explanations to
```

your initial bug report: a bug report should be to the point and
contain only human-verified information. Adding LLM-generated contents

> runs counter to those objectives, often even if the generated contents

Reviewer 3: I would put that `often` either before `runs` or drop it.

> Our advice is to avoid adding LLM-generated contents or explanations to
> your initial bug report: a bug report should be to the point and
> contain only human-verified information. Adding LLM-generated contents
> runs counter to those objectives, even if the generated contents
> looks useful to you.

Reviewer 4: ```suggestion
Our advice is to avoid adding LLM-generated content or explanations to
your initial bug report: a bug report should be to the point and
contain only human-verified information. Adding LLM-generated content
runs counter to those objectives, even if the generated content
looks useful to you.
```

> Indeed, judging the quality of a LLM generated contents about an

Reviewer: ```suggestion
Indeed, judging the quality of a LLM-generated content about an
```

unknown code base require a lot of efforts. Simultaneously, estimating
the faithfulness of LLM-generated contents without knowing your prior
understanding of the code base, nor your prompt, nor the model context

> also require a lot of efforts. Such efforts are in most cases better

Reviewer 3: I'd put effort in singular but IANANES.

> also require a lot of efforts. Such efforts are in most cases better

Reviewer 2: ```suggestion
unknown code base requires a lot of effort. Simultaneously, estimating
the faithfulness of LLM-generated contents without knowing your prior
understanding of the code base, nor your prompt, nor the model context
also requires a lot of effort. Such efforts are in most cases better
```

> Indeed, judging the quality of a LLM-generated content about an
> unknown code base requires a lot of effort. Simultaneously, estimating
> the faithfulness of LLM-generated contents without knowing your prior
> understanding of the code base, nor your prompt, nor the model context
> also requires a lot of effort. Such efforts are in most cases better
> spent investigating the bug itself.

Reviewer 4: Correcting the English:
```suggestion
Indeed, judging the quality of LLM-generated content about an
unknown code base requires a lot of effort. Likewise, for us, estimating
the faithfulness of LLM-generated contents without knowing your prior
understanding of the code base, nor your prompt, nor the model context
also requires a lot of effort. Such efforts are in most cases better
spent investigating the bug itself.
```
but I think this could simply be:

```suggestion
It can be a lot of effort - both for you creating the report, and for us
reviewing it - to assess the accuracy of an LLM-generated explanation
of a bug. We feel the effort in most cases is better spent investigating
the bug itself.
```

Author: My feeling is that your shorter version makes it less clear that the time potentially spent by the reporter to assess the accuracy of the LLM-generated contents may not reduce the time needed to fix the bug.
 
I think that this is a sign that the paragraph should reworded to better emphasize this dissymmetry.

Author: Or maybe another option would to focus on the human discussion aspect:

> It can be a lot of effort — both for you creating the report, and for us
reviewing it — to assess the accuracy of an LLM-generated explanation
of a bug. We feel the effort in most cases is better spent on reporting early
the bug and discussing it with other contributors.

Moreover, mixing human and LLM-generated contents in the same report
without a clear separation should be imperatively avoided. Indeed,

> the absence of a clear delineation between the two contents make it

Reviewer: ```suggestion
the absence of a clear delineation between the two contents makes it
```

> Moreover, mixing human and LLM-generated contents in the same report
> without a clear separation should be imperatively avoided. Indeed,
> the absence of a clear delineation between the two contents makes it
> very hard to ascertain the reliability of any part of the mixed
> contents.

Reviewer 4: ```suggestion
Moreover, mixing human and LLM-generated content in the same report
without a clear separation should be avoided. Indeed, the absence of a
clear delineation between the two makes it very hard to ascertain the
reliability of any part of the mixed content.
```

> We also advise some caution when shrinking reproduction case with
> LLMs. If you have any doubts about the LLM-shrinked reproduction case,
> please report both the original reproduction case and the shrinked
> one. In extreme cases, we have witnessed LLMs convincing people to
> reduce their bug report to a hallucination.

Reviewer 4: ```suggestion
We also advise some caution when shrinking reproduction cases using
LLMs. If you have any doubt about the LLM-shrunk reproduction case,
please report both the original reproduction case and the shrunk
one. In extreme cases, we have seen LLMs convincing people to
reduce their bug report to a hallucination!
```
